File: 3884-first-matching-character-from-both-ends/3884-first-matching-character-from-both-ends.cs
public class Solution {
    public int FirstMatchingIndex(string s) {
        var n = s.Length;
        int i = 0, j = n - 1;
        while (i <= j) {
            if (s[i++] == s[j--])
                return i - 1;
        }
        return -1;
    }
}

File: 3884-first-matching-character-from-both-ends/3884-first-matching-character-from-both-ends.java
class Solution {
    public int firstMatchingIndex(String s) {
        int n = s.length();
        int l = 0, r = n - 1;
        while (l <= r) {
            if (s.charAt(l++) == s.charAt(r--))
                return l - 1;
        }
        return -1;
    }
}